pennylane/measure.py

```diff
@@ -60,20 +54,15 @@ def circuit(x):
             "{} is not an observable: cannot be used with expval".format(op.name)
         )
 
-    if qml._current_context is not None:
-        # delete observables from QNode operation queue if needed
-        if isinstance(op, Tensor):
-            for o in op.obs:
-                _remove_if_in_queue(o)
-        else:
-            _remove_if_in_queue(op)
+    if isinstance(op, Tensor):
+        for o in op.obs:
+            qml.QueuingContext.remove_operator(o)
+    else:
+        qml.QueuingContext.remove_operator(op)
```

Comment:
I feel like this if statement could also be cleaned up, by having the Tensor class encapsulate queuing logic. But this is likely a separate PR.

Reply:
This is exactly what I proposed in parts B and C of the queuing ADR